src/io.c

```diff
@@ -211,9 +210,8 @@ nc_read(struct nc_session *session, char *buf, size_t count, uint32_t inact_time
             if (!interrupted) {
                 usleep(NC_TIMEOUT_STEP);
             }
-            nc_gettimespec_mono(&ts_cur);
-            if ((nc_difftimespec(&ts_cur, &ts_inact_timeout) < 1) || (nc_difftimespec(&ts_cur, ts_act_timeout) < 1)) {
-                if (nc_difftimespec(&ts_cur, &ts_inact_timeout) < 1) {
+            if ((nc_difftimespec_cur(&ts_inact_timeout) < 1) || (nc_difftimespec_cur(ts_act_timeout) < 1)) {
+                if (nc_difftimespec_cur(&ts_inact_timeout) < 1) {
```

Comment:
How is this change related to adding support for PAM? If a timer rework is required, let's do that in a separate commit, please.

Reply:
My apologies, this was not supposed to be an actual pull request, hence the .so. Anyways, I appreciate your advice.